File: flutter_ui/lib/models/job_template.dart
class JobTemplate {
  final String id;
  final String name;
  final String description;
  final String prompt;
  final List<String> skillIds;
  final String? workspaceId;
  final String? model;
  final int? timeoutSecs;
  final int priority;
  final List<String> tags;
  final String createdAt;
  final String updatedAt;

  JobTemplate({
    required this.id,
    required this.name,
    this.description = '',
    required this.prompt,
    this.skillIds = const [],
    this.workspaceId,
    this.model,
    this.timeoutSecs,
    this.priority = 5,
    this.tags = const [],
    required this.createdAt,
    required this.updatedAt,
  });

  factory JobTemplate.fromJson(Map<String, dynamic> json) => JobTemplate(
        id: json['id'] ?? '',
        name: json['name'] ?? '',
        description: json['description'] ?? '',
        prompt: json['prompt'] ?? '',
        skillIds: List<String>.from(json['skill_ids'] ?? []),
        workspaceId: json['workspace_id'],
        model: json['model'],
        timeoutSecs: json['timeout_secs'] as int?,
        priority: json['priority'] ?? 5,
        tags: List<String>.from(json['tags'] ?? []),
        createdAt: json['created_at'] ?? '',
        updatedAt: json['updated_at'] ?? '',
      );
}

File: crates/claw-models/src/job_template.rs
use chrono::{DateTime, Utc};
use serde::{Deserialize, Serialize};
use uuid::Uuid;

use crate::OutputDest;

/// A reusable job definition that can be referenced by jobs, crons, and pipelines.
#[derive(Debug, Clone, Serialize, Deserialize)]
pub struct JobTemplate {
    pub id: Uuid,
    pub name: String,
    #[serde(default)]
    pub description: String,
    pub prompt: String,
    #[serde(default)]
    pub skill_ids: Vec<String>,
    #[serde(default)]
    pub tool_ids: Vec<String>,
    pub workspace_id: Option<Uuid>,
    pub model: Option<String>,
    pub timeout_secs: Option<u64>,
    pub allowed_tools: Option<Vec<String>>,
    #[serde(default)]
    pub output_dest: OutputDest,
    #[serde(default)]
    pub tags: Vec<String>,
    #[serde(default = "default_template_priority")]
    pub priority: u8,
    pub created_at: DateTime<Utc>,
    pub updated_at: DateTime<Utc>,
}

fn default_template_priority() -> u8 {
    5
}

#[derive(Debug, Clone, Serialize, Deserialize)]
pub struct CreateJobTemplateRequest {
    pub name: String,
    #[serde(default)]
    pub description: String,
    pub prompt: String,
    #[serde(default)]
    pub skill_ids: Vec<String>,
    #[serde(default)]
    pub tool_ids: Vec<String>,
    pub workspace_id: Option<Uuid>,
    pub model: Option<String>,
    pub timeout_secs: Option<u64>,
    pub allowed_tools: Option<Vec<String>>,
    #[serde(default)]
    pub output_dest: OutputDest,
    #[serde(default)]
    pub tags: Vec<String>,
    pub priority: Option<u8>,
}
